scripts/linter_ignores.py

```diff
@@ -0,0 +1,73 @@
+"""
+Check for instances of linter directive ignores without comments.
+
+In order to keep code quality high, we want to avoid future additions of
+noqa or type: ignore directives without comments explaining why they are
+necessary.
+"""
+
+from __future__ import annotations
+
+import pathlib
+import re
+import sys
+from collections.abc import Iterator
+
+TYPE_IGNORE_COMMENT_REGEX: re.Pattern[str] = re.compile(
+    r" # type: ignore(?P<code>\[.*?\])?(?P<comment>\s*# .*)?$"
+)
+NOQA_IGNORE_COMMENT_REGEX: re.Pattern[str] = re.compile(
+    r" # noqa: (?P<code>(\s*,)?.*?)*(?P<comment>\s*# .*)?$"
```

Comment:
Actually I think I recommended the wrong thing and you can get rid of `(\s*,)?`. It should be `(,\s*)` but changing the `?` to a `*` outside the parentheses is enough since the match is against `.` (ie anything).